src/tests/test_tls_stream_integration.cpp

```diff
@@ -0,0 +1,722 @@
+/*
+* TLS ASIO Stream Client-Server Interaction Test
+* (C) 2018-2020 Jack Lloyd
+*     2018-2020 Hannes Rantzsch, Rene Meusel
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include "tests.h"
+
+#if defined(BOTAN_HAS_TLS) && defined(BOTAN_HAS_TLS_ASIO_STREAM)
```

Comment:
You'll also want `BOTAN_TARGET_OS_HAS_THREADS` so this test is disabled on baremetal, IncludeOS etc